src/main/java/seedu/address/model/person/Class.java

```diff
@@ -3,42 +3,148 @@
+    public Class(LocalDate date, LocalTime startTime, LocalTime endTime, String classDateTime) {
         this.date = date;
         this.startTime = startTime;
         this.endTime = endTime;
+        this.classToString = convertToString(date, startTime, endTime);
+        this.classDateTime = classDateTime;
+    }
+
+    /**
+     * Returns a formatted date and time.
+     *
+     * @param date LocalDate object.
+     * @param startTime LocalTime object.
+     * @param endTime LocalTime object.
+     * @return String.
+     */
+    public static String convertToString(LocalDate date, LocalTime startTime, LocalTime endTime) {
+        return convertToDateString(date) + " " + convertToTimeString(startTime, endTime);
+    }
+
+    /**
+     * Returns a formatted date.
+     *
+     * @param date LocalDate object.
+     * @return String.
+     */
+    public static String convertToDateString(LocalDate date) {
+        return date.format(DateTimeFormatter.ofPattern("dd MMM yyyy"));
+    }
+
+    /**
+     * Returns a formatted time.
+     *
+     * @param startTime LocalTime object.
+     * @param endTime LocalTime object.
+     * @return String.
+     */
+    public static String convertToTimeString(LocalTime startTime, LocalTime endTime) {
+        String time = "";
+        int startHour = startTime.getHour();
+        int startMin = startTime.getMinute();
+        int endHour = endTime.getHour();
+        int endMin = endTime.getMinute();
+        if (startHour >= 12) {
+            if (startHour != 12) {
+                startHour -= 12;
+            }
+            if (startMin == 0) {
+                time += startHour + "PM";
+            } else if (startMin < 10) {
+                //Pad with zero
+                time += startHour + ".0" + startMin + "PM";
+            } else {
+                time += startHour + "." + startMin + "PM";
+            }
+        } else {
+            if (startMin == 0) {
+                time += startHour + "AM";
+            } else if (startMin < 10) {
+                //Pad with zero
+                time += startHour + ".0" + startMin + "AM";
+            } else {
+                time += startHour + "." + startMin + "AM";
+            }
+        }
+        time += "-";
+        if (endHour >= 12) {
+            if (endHour != 12) {
+                endHour -= 12;
+            }
+            if (endMin == 0) {
+                time += endHour + "PM";
+            } else if (endMin < 10) {
+                //Pad with zero
+                time += endHour + ".0" + endMin + "PM";
+            } else {
+                time += endHour + "." + endMin + "PM";
+            }
+        } else {
+            if (endMin == 0) {
+                time += endHour + "AM";
+            } else if (endMin < 10) {
+                //Pad with zero
+                time += endHour + ".0" + endMin + "AM";
+            } else {
+                time += endHour + "." + endMin + "AM";
+            }
+        }
```

Comment:
The parsing process for `startTime` and `endTime` is effectively the same and hence should be abstracted out by writing a helper method to avoid duplicate code.